backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,321 @@
 package util
 
 import (
+	"context"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
 	"github.com/redhat-appstudio/managed-gitops/backend-shared/config/db"
 	"github.com/stretchr/testify/assert"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logger "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+var _ = Describe("Test utility functions.", func() {
+	Context("It should execute and test GetOrCreateManagedEnvironmentByNamespaceUID function.", func() {
+
+		It("Should create new managedEnvironment and other resources, if called second time then it should return existing resources.", func() {
+
+			workSpaceUid := uuid.NewUUID()
+			ctx := context.Background()
+
+			dbQueries, err := db.NewUnsafePostgresDBQueries(false, false)
+			Expect(err).To(BeNil())
+			defer dbQueries.CloseDatabase()
+
+			workspace := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-user",
+					UID:       workSpaceUid,
+					Namespace: "test-namespace",
+				},
+				Spec: v1.NamespaceSpec{},
+			}
+
+			// ----------------------------------------------------------------------------
+			By("It should create new ManagedEnvironment for given namespace.")
+			// ----------------------------------------------------------------------------
+			log := log.FromContext(context.Background())
+
+			managedEnvironment, isNew, err := GetOrCreateManagedEnvironmentByNamespaceUID(ctx, workspace, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Verify that other database entries are also created.")
+			// ----------------------------------------------------------------------------
+
+			// Check KubernetesToDBResourceMapping resource
+			kubernetesToDBResourceMappingget := db.KubernetesToDBResourceMapping{
+				KubernetesResourceType: "Namespace",
+				KubernetesResourceUID:  string(workSpaceUid),
+				DBRelationType:         "ManagedEnvironment",
+			}
+
+			err = dbQueries.GetDBResourceMappingForKubernetesResource(ctx, &kubernetesToDBResourceMappingget)
+			Expect(err).To(BeNil())
+
+			// Check ClusterCredentials resource
+			clusterCredentials := db.ClusterCredentials{
+				Clustercredentials_cred_id: managedEnvironment.Clustercredentials_id,
+			}
+			err = dbQueries.GetClusterCredentialsById(ctx, &clusterCredentials)
+			Expect(err).To(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("If called 2nd time, it should return existing ManagedEnvironment instead of creating new and flag should be False.")
+			// ----------------------------------------------------------------------------
+			managedEnvironmentSecond, isNewSecond, errSecond := GetOrCreateManagedEnvironmentByNamespaceUID(ctx, workspace, dbQueries, log)
```

Comment:
I think `isNew` as a prev variable name will suffice for this as well, do we have a use-case of using `isNewSecond` separately?